aspnetcore/security/authentication/mfa.md

```diff
@@ -0,0 +1,560 @@
+
+This article covers the following:
+
+* What is MFA and what MFA flows are recommended
+* Configure MFA for administration pages using ASP.NET Core Identity
+* Send MFA signin requirement to OpenID Connect server 
+* Force ASP.NET Core OpenID Connect client to require MFA
+
+## MFA, 2FA
+
+MFA requires at least two or more types of proof for an identity like something you know, something you possess, or something you inherit for the user to authenticate.
+
+2FA is like a subset of MFA,but the difference being that MFA can require 2 or more factors to prove the identity.
+
+### MFA TOTP (Time-based One-time Password Algorithm)
+
+Multi-factor authentication using TOTP is a supported implementation using ASP.NET Core Identity. This can be used together with the following Apps:
+
+- Microsoft Authenticator App
+- Google Authenticator App
+
+See the following link for implementation details:
+
+[Enable QR Code generation for TOTP authenticator apps in ASP.NET Core](xref:security/authentication/identity-enable-qrcodes)
+
+### MFA FIDO2 or Passwordless
+
+FIDO2 is the most secure way of doing Multi-factor authentication and is the only MFA flow which protects against phishing attacks. At present ASP.NET Core does not support FIDO2 directly. FIDO2 can be used for MFA or passwordless flows.
+
+Azure Active Directory provides support for FIDO2 and passwordless flows.
+
+[Passwordless authentication options for Azure Active Directory](https://docs.microsoft.com/en-us/azure/active-directory/authentication/concept-authentication-passwordless)
+
+You can implement ASP.NET Core with FIDO2 by using the following OSS FIDO2 implementation:
+
+[FIDO2 .NET library for FIDO2 / WebAuthn Attestation and Assertion using .NET](https://github.com/abergs/fido2-net-lib)
+
+### MFA SMS
+
+Although MFA with SMS increases security massively compared with password authentication (Single factor), it is no longer recommended to use SMS as a second factor as too many known attack vectors exist for this type of implementation.
+
+## Configure MFA for administration pages using ASP.NET Core Identity
+
+MFA could be forced on users to access sensitive pages within an ASP.NET Core Identity application. This could be useful for applications where different levels of access exist for the different identities. For example, users might be able to view the profile data using a password login, but an administrator would be required to use MFA to access the admin pages.
+
+### Extending the Login with a MFA claim
+
+The demo code is setup using ASP.NET Core with Identity and Razor Pages. The AddIdentity method is used instead of AddDefaultIdentity one, so an IUserClaimsPrincipalFactory implementation can be used to add claims to the identity after a successful login.
```

Comment:
```suggestion
The demo code is setup using ASP.NET Core with Identity and Razor Pages. The `AddIdentity` method is used instead of `AddDefaultIdentity` one, so an `IUserClaimsPrincipalFactory` implementation can be used to add claims to the identity after a successful login.
```